### PR TITLE
Almost completed branch hours view

### PR DESCRIPTION
**Pull request by jerometujague**
The branch hours view is almost completed, everything shows up correctly, but I am running into a snag when I try to filter or edit the day of the week. I think it has something to do with how I am inputting the value into the filter options or new edit items, but I am not sure.
Would you mind taking a look?

### CODE REVIEW

File: src/main/resources/assets/src/managerView/branchHoursView.js [hunk 1]
```diff
@@ -1,13 +1,283 @@
...
+        })
+    }
+
+    getBranchName(id) {
+        return this.props.branches.find(b => { return b.id == id }).name;
+    }
+
+    convertIntToDay(dayNum) {
+        switch(dayNum){
+            case 1:
```
Comment (jmbockhorst) [inline src/main/resources/assets/src/managerView/branchHoursView.js]: I think 1 start with Monday.

**Comment by jerometujague** [comment]
Also, once this is working properly I'm going to pull the convert Day to Int and Int to Day functions out into the functions.js page.

File: src/main/resources/assets/src/managerView/branchHoursView.js [hunk 2]
```diff
@@ -1,13 +1,283 @@
...
 
     render() {
+        const headerNames = ['Open Time', 'Close Time', 'Branch', 'Day Of Week'];
+
+        // Filter options
+        const filterOptions = [
+            getTopResults(this.props.branchHours.map(a => a.openTime[0])).map(r => convertTime24to12(r.item)),
+            getTopResults(this.props.branchHours.map(a => a.closeTime[0])).map(r => convertTime24to12(r.item)),
+            getTopResults(this.props.branchHours.map(b => b.branchId)).map(r => this.getBranchName(r.item)),
+            getTopResults(this.props.branchHours.map(h => h.dayOfWeek)).map(r => this.convertIntToDay(r.item))];
```
Comment (jmbockhorst) [inline src/main/resources/assets/src/managerView/branchHoursView.js]: I beleive that `r.item` here is a string. In `convertIntToDay` it is checking against integers. So either convert it to integers or check for strings in the function. You can use `parseInt()`.

**Comment by jmbockhorst** [comment]
> Also, once this is working properly I'm going to pull the convert Day to Int and Int to Day functions out into the functions.js page, unless you don't think they need to be moved since they're only used in this view.

Yeah moving them would be great.

File: src/main/resources/assets/src/managerView/branchHoursView.js [hunk 3]
```diff
@@ -1,13 +1,283 @@
...
+
+        const validOpenTime = openTime.match('^(0[0-9]|1[0-9]|2[0-3]|[0-9]):[0-5][0-9]$');
+        const validCloseTime = closeTime.match('^(0[0-9]|1[0-9]|2[0-3]|[0-9]):[0-5][0-9]$');
+
+        // Check if the data is valid
+        if (!validOpenTime || !validCloseTime || !branch || !dayOfWeek) {
+            const newEditorData = this.state.editorData;
+
+            if (!validOpenTime) {
+                newEditorData[0] = "Invalid open time";
```
Comment (jmbockhorst) [inline src/main/resources/assets/src/managerView/branchHoursView.js]: These should be: `newEditorData.editErrors[0]`. I realized I messed that up too for some of them.

File: src/main/resources/assets/src/managerView/branchHoursView.js (hunk 1 above)
Comment (jerometujague) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: Oh my bad, I was basing it off of the calendar table's Sunday through Saturday 1-7 format.

Comment (jerometujague) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: I know we talked about this but I can't remember what we decided, why are the days as ints for branch hours different from the calendar days?

File: src/main/resources/assets/src/managerView/branchHoursView.js (hunk 3 above)
Comment (jerometujague) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: Got you. Update these now.

File: src/main/resources/assets/src/managerView/branchHoursView.js (hunk 1 above)
Comment (jmbockhorst) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: I’m not sure I understand. Branch hours has day of week for weekly hours. Calendar days or id is from 1-365

Comment (jerometujague) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: Sorry, I meant in the calendar table there is a column called Day_Of_Week and it uses 1-7 starting with Sunday. I guess the two tables don't really interact with each other so it doesn't matter.

Comment (jmbockhorst) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: We don’t use that column.

Comment (jerometujague) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: Exactly, I was more just curious for the sake of uniformity. But it doesn't matter either way.

File: src/main/resources/assets/src/managerView/branchHoursView.js (hunk 2 above)
Comment (jerometujague) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: Okay, the editing works fine now, but when I go to filter by Day of Week, it doesn't do anything. Is this done through the filterOptions const and am I inputting the wrong type of data?

Comment (jmbockhorst) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: The actually filtering is done on lines 243-248.

Comment (jerometujague) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: Ok. Not sure what is going wrong, the filtering is working for everything except day of week.

Comment (jmbockhorst) [reply src/main/resources/assets/src/managerView/branchHoursView.js]: Because you are doing table length - 1 in the for loop

**Pull request merged**